File: fixtures/dart/type_inference/overloads/overloads.dart
class Overloads {
  int combine(int a, int b) => a + b;

  String combineStrings(String a, String b) => a + b;

  String combineMixed(int a, String b) => '$a$b';

  String run() {
    final ints = combine(1, 2);
    final strs = combineStrings('a', 'b');
    final mixed = combineMixed(1, 'b');
    return '$ints$strs$mixed';
  }
}

File: fixtures/php/type_inference/overloads/Overloads.php
<?php

class Overloads {
    public function combineInts(int $a, int $b): int {
        return $a + $b;
    }

    public function combineStrings(string $a, string $b): string {
        return $a . $b;
    }

    public function combineMixed(int $a, string $b): string {
        return $a . $b;
    }

    public function run(): string {
        $ints = $this->combineInts(1, 2);
        $strs = $this->combineStrings("a", "b");
        $mixed = $this->combineMixed(1, "b");
        return $ints . $strs . $mixed;
    }
}
